relay-server/src/actors/events.rs

```diff
@@ -769,12 +804,14 @@ impl EventProcessor {
         let apple_crash_report_attachment = envelope
             .get_item_by(|item| item.attachment_type() == Some(AttachmentType::AppleCrashReport));
 
+        let mut event = state.event.get_or_insert_with(Event::default);
         if let Some(item) = minidump_attachment {
             state.metrics.bytes_ingested_event_minidump = Annotated::new(item.len() as u64);
-            self.write_native_placeholder(&mut state.event, true);
+            self.write_native_placeholder(&mut event, true);
+            Self::write_minidump_timestamp(&mut event, item);
```

Comment:
maybe just make it a free function then? Its a bit unconventional to have static functions that are not constructors.
Maybe instead of taking a `&mut Event`, rename the function to `try_extract_timestamp` or similar and return an `Option<DateTime<Utc>>`.

Reply:
Since all functions are on `EventProcessor` for the time, I'd suggest to just give it an unused `&self`. In one of the upcoming changes we will likely have to access the `config`, so it's totally fine to have it on there. I think taking `&mut Event` is fine in this case.